dlt/destinations/snowflake/configuration.py

```diff
@@ -23,21 +23,24 @@ def _read_private_key(private_key: str, password: Optional[str] = None) -> bytes
         from cryptography.hazmat.primitives.asymmetric import rsa
         from cryptography.hazmat.primitives.asymmetric import dsa
         from cryptography.hazmat.primitives import serialization
+        from cryptography.hazmat.primitives.asymmetric.types import PrivateKeyTypes
     except ModuleNotFoundError as e:
         raise MissingDependencyException("SnowflakeCredentials with private key", dependencies=[f"{version.DLT_PKG_NAME}[snowflake]"]) from e
 
+    load_key_func = serialization.load_pem_private_key
+    private_key_bytes_decoded = private_key.encode()
     try:
         # decode base64 encoded private key
-        private_key_decoded = base64.b64decode(private_key).decode('utf-8')
+        private_key_bytes_decoded = base64.b64decode(private_key)
+        load_key_func = serialization.load_der_private_key
```

Comment:
again the PEM key also decodes as base64 so you will assign der key func to it. see CI - it fails
what I'd do here: just load the key as DER, if that fails - load as PEM in exception handler